src/adm/filter.rs

```diff
@@ -258,6 +263,14 @@ impl AdmFilter {
                     return None;
                 }
 
+                if device_info.ff_version < 91
+                    && !self.legacy_list.contains(&tile.name.to_lowercase())
```

Comment:
The legacy status of the requesting browser should also be included in the cache key (`AudienceKey`) since it affects the Tiles content.

Reply:
Good catch. This would add a multiplier to the current cache keyspace.

Shall we add the status to the cache key (multiplier: 2), or add the ff_version (multiplier: N)?

Reply:
I think a boolean flag of legacy or not is all we need/want.

Reply:
Way ahead of y'all. I added a function to `DeviceInfo` to return if we should provide legacy only. 

I was debating if the cache should store the first pass validated ADM result rather than the finalized list to provide a pool of data we can progressively filter later, but that's YAGNI.